pkg/api/api.go

```diff
@@ -279,8 +279,8 @@ func (hs *HTTPServer) registerRoutes() {
 		}, reqOrgAdmin)
 
 		apiRoute.Get("/frontend/settings/", hs.GetFrontendSettings)
-		apiRoute.Any("/datasources/proxy/:id/*", reqSignedIn, hs.ProxyDataSourceRequest)
-		apiRoute.Any("/datasources/proxy/:id", reqSignedIn, hs.ProxyDataSourceRequest)
+		apiRoute.Any("/datasources/proxy/:id/*", reqSignedIn, hs.DataProxy.ProxyDataSourceRequest)
```

Comment:
If the routes are going to be registered here I would prefer having the request handled (`ProxyDataSourceRequest`) in the api package rather than in the datasourceproxy package.

As an alternative you can register the routes in the service itself. For example see 
https://github.com/grafana/grafana/blob/d0fa7b81f0ff4803a538455d7df04d2410217244/pkg/services/ngalert/ngalert.go#L41
https://github.com/grafana/grafana/blob/d0fa7b81f0ff4803a538455d7df04d2410217244/pkg/services/ngalert/api/api.go#L42

Reply:
We intend to use these functions from other consumers and thus don't want to register them here to prevent a circular import. I've created a shallow passthrough function on the httpserver as a middle ground here. Does this work?